server/mscalendar/user.go

```diff
@@ -137,6 +138,25 @@ func (m *mscalendar) DisconnectUser(mattermostUserID string) error {
 		return err
 	}
 
+	// Unlink events owned by the user that is disconnecting its account
+	linkedEventsLeft := make(map[string]string)
+	for eventID, channelID := range storedUser.ChannelEvents {
+		if errStore := m.Store.DeleteLinkedChannelFromEvent(eventID, channelID); errStore != nil {
+			linkedEventsLeft[eventID] = channelID
+		}
+	}
+	if len(linkedEventsLeft) != 0 {
+		storedUser.ChannelEvents = linkedEventsLeft
+		if errStore := m.Store.StoreUser(storedUser); errStore != nil {
+			m.Logger.With(bot.LogContext{
+				"err":                  errStore,
+				"mm_user_id":           storedUser.MattermostDisplayName,
+				"linked_channels_left": linkedEventsLeft,
+			}).Errorf("error storing user after failing deleting linked channels from store")
+		}
+		return fmt.Errorf("error deleting linked channels from events")
```

Comment:
This should be in the `if errStore != nil` block right?

Reply:
Not really, this will show an error either way if there are [events pending from removal](https://github.com/mattermost/mattermost-plugin-mscalendar/pull/296/files#diff-2c556af43018651b32cd1e69d78e9654556352a9ccc7446aa398a7884486f1f8R143-R147) in the previous block. Note how we only log the error on store but actually return an error on that condition.

- We try to remove events, if an event fails deletion (unlink) we add it to a `map`.
- After that, we check the `map`, if it contains elements, we failed to remove something we should **return** `error`.
  - While in this condition, we store the linked channels that are pending removal back to the user, and if that fails the **log** an error.